lib/connectionContext.ts

```diff
@@ -75,17 +86,54 @@ export namespace ConnectionContext {
    */
   export const userAgent: string = "/js-event-hubs";
 
-  export function create(config: ConnectionConfig, tokenProvider?: TokenProvider): ConnectionContext {
+  function encodeMessage(body: any): any {
+    let result: any;
+    if (body != undefined)
+      if (isBuffer(body))
+        result = rhea.message.data_section(body);
+      else if (typeof body !== "string")
+        result = rhea.message.data_section(Buffer.from(JSON.stringify(body), "utf8"));
+      else
+        result = rhea.message.data_section(Buffer.from(body, "utf8"));
+    else
+      result = rhea.message.data_section(Buffer.from("", "utf8"));
+
+    return result;
+  }
+
+  function decodeMessage(body: any): any {
+    let processedBody: any = body;
+    if (isBuffer(body)) {
+      processedBody = body.toString("utf8");
+    } else if (typeof body === "string") {
+      processedBody = body;
+    } else if (body.content) {
+      processedBody = body.content.toString("utf8");
+    }
+    try {
+      processedBody = JSON.parse(processedBody);
+    } catch (err) {
```

Comment:
>err [](start = 13, length = 3)

should you catch only SyntaxError? what other errors should you not catch?

Reply:
We should definitely log the syntax error. 

> what other errors should you not catch?

I don't see any other type of errors happening in the above lines in the function since all the statements are guarded by the if condition. However, it is worth while to have an uber try catch, log the error and return the input parameter `body` as-is.